packages/nimble-components/src/menu/template.ts

```diff
@@ -0,0 +1,30 @@
+import {
+    children,
+    html,
+    slotted,
+    type ViewTemplate
+} from '@microsoft/fast-element';
+import type { FoundationElementTemplate } from '@microsoft/fast-foundation';
+import type { Menu } from '.';
+
+/* eslint-disable @typescript-eslint/indent */
+// prettier-ignore
+export const template: FoundationElementTemplate<
+ViewTemplate<Menu>
+> = () => html`
+    <template
+        slot="${x => {
+            if (x.slot) {
+                return x.slot;
+            }
+            return x.isNestedMenu() ? 'submenu' : '';
+        }}"
+        role="menu"
+        @keydown="${(x, c) => x.handleMenuKeyDown(c.event as KeyboardEvent)}"
+        @focusout="${(x, c) => x.handleFocusOut(c.event as FocusEvent)}"
+        ${children({ property: 'itemIcons', subtree: true, selector: ':scope > * > [slot="start"]' })}
```

Comment:
For this to be reliable we would need to include an attrbuteFilter for slot as that can change programmatically at a different time than dom insertion / removal: https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/observe#attributefilter